### DISCUSSION TITLE
Misunderstanding links?

### DISCUSSION THREAD

**Discussion by Madd0g** [pnpm/pnpm#5476] [Q&A]
Hey,

I'm on macos, I'm trying to understand why subsequent installs take up to 10 seconds.
I'm using `prefer-offline` and I tried using both `package-import-method=auto` and `hardlink`.

I found the `store-dir` which has lots of folders with lots of files.

I found that under `my-project/node_modules`, all modules are linked to `my-project/node_modules/.pnpm` (virtual store).

But I'm wondering - shouldn't the modules under the virtual store `.pnpm` be linked to the `store-dir` (the global one)? Everything under `.pnpm` looks like real files.

There a note in the docs about macos, is that what it's about? If that feature worked the files would've been "cloned" but not "copied"? and hardlink only makes a difference between node_modules and the virtual store?

> Even though macOS supports cloning, there is currently [a bug in Node.js](https://github.com/libuv/libuv/pull/2578) that prevents us from using it in pnpm. If you have ideas how to fix it, [help us](https://github.com/pnpm/pnpm/issues/5001).

Thank you for helping me understand

**Comment by zkochan** [comment]
What do you mean by sunsequent install? Just running `pnpm install` again with no changes? Or like switching to a different branch and running `pnpm install`?

**Reply by Madd0g** [reply]
I mean running install once then deleting node_modules and running install again, with the same package.json and lockfile. Or doing it from another folder entirely with same package/lockfile.
